source/retryable-writes/tests/insertOne-errorLabels.yml

```diff
@@ -0,0 +1,80 @@
+runOn:
+    -
+        minServerVersion: "4.3.0"
+        topology: ["replicaset", "sharded"]
+
+data: []
+
+tests:
+    -
+        description: "InsertOne succeeds after RetryableWriteError from server"
+        failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["insert"]
+                errorLabels: ["RetryableWriteError"]
+                errorCode: 112 # WriteConflict, not normally a retryable error code
+            operation:
+                name: "insertOne"
+                arguments:
+                    document: { _id: 1, x: 11 }
+            outcome:
+                result:
+                    insertedId: 1
+                collection:
+                    data:
+                        - { _id: 1, x: 11 }
+    -
+        description: "InsertOne succeeds after a connection failure"
+        failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["insert"]
+                closeConnection: true
+        operation:
+            name: "insertOne"
+            arguments: { _id: 1, x: 11 }
+        outcome:
+            result:
+                insertedId: 1
+            collection:
+                data:
+                    - { _id: 1, x: 11 }
+    -
+        description: "InsertOne fails with a RetryableWriteError label after two connection failures"
+        failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+                failCommands: ["insert"]
+                closeConnection: true
+        operaion:
+            name: "insertOne"
+            arguments:
+                document: { _id: 1, x: 11 }
+        outcome:
+            error: true
+            result:
+                errorLabelsContain: ["RetryableWriteError"]
+            collection:
+                data: []
+
+    -
+        description: "InsertOne does not retry and fails on error without RetryableWriteError label"
+        failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["insert"]
+                errorLabels: []
```

Comment:
My first suggestion: Rather than use `errorLabels` here, can we simply use an error code that we know will not be labeled by either the driver or server (e.g. 112, WriteConflict) and move this to `insertOne-serverErrors.yml`?

If we do that, this might be redundant since we already have "InsertOne fails after Interrupted" in that file, which asserts that an insert fails after the first attempt receives a non-retryable error. In that case, I think we can delete this entirely and simply add an `errorLabelsOmit` assertion to that existing test.

If this PR is mainly going to focus on `insertOne` test files before we settle on changes to make across the board, I think you should take a look at all tests in `insertOne-serverErrors.yml` and `insertOne.yml` that have error assertions and see where you can add `errorLabelsContain` or `errorLabelsOmit` assertions. Those should be portable across all server versions, since the driver and server labeling should be consistent. Furthermore, any failures in such tests down the line should inform us of a possible deviation as soon as it pops up in a new server release.

Reply:
I like that! Thank you.

Reply:
I like that! Thank you!